ui/src/features/Editor/useCanvasCopyPaste.ts

```diff
@@ -120,6 +128,8 @@ export default ({
             newSubworkflowNodes,
             newSubworkflowEdges,
           );
+
+          pastedWorkflows.push(newSubworkflow);
```

Comment:
I think this is mutating the workflows array being passed into this function, which is not good. 

It is better to create a newWorkflowCreation and move the workflow logic there, returning the newowrkflow and copying that along with the newly created nodes and edges. Keeps everything same-y and avoids confusing mutations. 🙏🏻